applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing_matrix_free.h

```diff
@@ -142,42 +142,53 @@ class MapperVertexMorphingMatrixFree : public Mapper
         mValuesDestination[2].clear();
 
         // Perform mapping
-        for(auto& node_i : mrDestinationModelPart.Nodes())
+        const auto destination_nodes_begin = mrDestinationModelPart.NodesBegin();
+
+        #pragma omp parallel for firstprivate(destination_nodes_begin)
+        for(IndexType node_itr=0; node_itr<mrDestinationModelPart.NumberOfNodes(); node_itr++)
         {
+            auto node_i = destination_nodes_begin + node_itr;
```

Comment:
if you try to obtain a reference here, you don't need to change the rest of the code